django/sierra/base/models.py

```diff
@@ -1948,6 +1947,68 @@ def get_call_numbers(self):
                 call_number_tuples.extend(call_numbers)
         return call_number_tuples
 
+    def get_shelving_call_number(self):
+        """
+        Returns one tuple that should represent the call number used
+        to shelve this item: (call_number, cn_type).
+        """
+        call_number, cn_type = (None, None)
+        item_cn_tuples = self.get_call_numbers()
+        bib_cn_tuples = []
+        try:
+            bib_cn_tuples = (self.bibrecorditemrecordlink_set.all()[0]
+                                .bib_record.get_call_numbers())
+        except IndexError:
+            pass
+
+        # First: The first cn_tuple on the item record OR, if that
+        # doesn't exist, the first cn_tuple on the bib record
+        # is the shelving call number.
+
+        if len(item_cn_tuples) > 0:
+            (call_number, cn_type) = item_cn_tuples[0]
+        elif len(bib_cn_tuples) > 0:
+            (call_number, cn_type) = bib_cn_tuples[0]
+
+        # Second: Check the main shelving call number against any Gov
+        # Docs numbers that are on the bib. If the call number string
+        # matches a Gov Docs number, then the type should be "sudoc" no
+        # matter what.
+
+        if call_number and (call_number, 'sudoc') in bib_cn_tuples:
+            cn_type = 'sudoc'
+
+        # Third: Periodical call numbers. We might have a periodical
+        # shelved by title: if it has an ITYPE code of 5 (bound
+        # periodicals) and the call number is just two letters (like
+        # MT), or if the word "periodical" appears in the call number,
+        # or if there's ANY c-tagged field on the item with "SHELVED BY
+        # TITLE." In these cases, we get the Bib title and append it to
+        # the end of the call number.
+
+        probably_shelved_by_title = (
+            self.itype.code_num == 5 and (not call_number
+                or re.search(r'^[A-Za-z]{,2}$', call_number))
+            or (call_number and 
+                re.search(r'^periodical', call_number, re.IGNORECASE)))
+
+        if not probably_shelved_by_title:
+            item_vfs = self.record_metadata.varfield_set.all()
+            for cn in helpers.get_varfield_vals(item_vfs, 'c', many=True,
+                    content_method='display_field_content'):
+                if cn.strip().upper() == 'SHELVED BY TITLE':
+                    probably_shelved_by_title = True
+                    break
+
+        if probably_shelved_by_title:
+            bib = self.bibrecorditemrecordlink_set.all()[0].bib_record
+            title = bib.bibrecordproperty_set.all()[0].best_title
+            title = re.sub(r'\.*\s*$', r'', title)
+            call_number = '{} -- {}'.format(call_number, title)
+            cn_type = 'other'
+        
```

Comment:
I have a feeling my code is littered with this. I should probably have considered starting using a pep8 linter long ago. Thanks for pointing it out.